website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,461 @@
+  to   = aws_instance.b["tiny"]
+}
+
+# The old configuration used "count" and the new configuration
+# uses "for_each", with the following mappings from
+# index to key:
+moved {
+  from = aws_instance.c[0]
+  to   = aws_instance.c["small"]
+}
+moved {
+  from = aws_instance.c[1]
+  to   = aws_instance.c["tiny"]
+}
+
+# The old configuration used "count" and the new configuration
+# uses neither "count" nor "for_each", and we want to keep
+# only the object at index 2.
+moved {
+  from = aws_instance.d[2]
+  to   = aws_instance.d
+}
+```
+
+## Renaming a Module Call
+
+You can rename a call to a module in a similar way as renaming a resource.
+Consider the following original module version:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+When applying this configuration, Terraform would prefix the addresses for
+any resources declared in this module with the module path `module.a`.
+For example, a resource `aws_instance.example` would have the full address
+`module.a.aws_instance.example`.
+
+If you later choose a better name for this module call, then you can change the
+name label in the `module` block and record the old name inside a `moved` block:
+
+```hcl
+module "b" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+When creating the next plan for each configuration using this module, Terraform
+will treat any existing object addresses beginning with `module.a` as if
+they had instead been created in `module.b`. `module.a.aws_instance.example`
+would be treated as `module.b.aws_instance.example`.
+
+Both of the addresses in this example referred to a module call as a whole, and
+so Terraform recognizes the move for all instances of the call. If this
+module call used `count` or `for_each` then it would apply to all of the
+instances, without the need to specify each one separately.
+
+## Enabling `count` or `for_each` For a Module Call
+
+Imagine that the initial version of your module contains a single-instance
+module:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+Applying this configuration would cause Terraform to create objects whose
+addresses begin with `module.a`.
+
+Later requirements might lead to you needing to use
+[`count`](../../meta-arguments/count.html) with this resource, in order
+to systematically declare multiple instances. To preserve an object that
+was previously associated with `aws_instance.a` alone, you can used a `moved`
+block to specify which instance key that object will take in the new
+configuration:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+  count  = 3
+
+  # (module arguments)
+}
+
+moved {
+  from = module.a
+  to   = module.a[2]
+}
+```
+
+The above will cause Terraform to treat all objects in `module.a` as if they
+were originally created in `module.a[2]`, and thus it will plan to create
+new objects only for `module.a[0]` and `module.a[1]`.
```

Comment:
Going to keep the "will" here as we're describing something that is truly future from the example. 

```suggestion
The configuration above directs Terraform to treat all objects in `module.a` as if they were originally created in `module.a[2]`. As a result, Terraform plans to create new objects only for `module.a[0]` and 'module.a[1]`.
```